CONTRIBUTING.md

```diff
@@ -1,7 +1,188 @@
+# Contributing
+
 In order for us to consider merging a contribution, you will need to sign our
 **C**ontributor **L**icense **A**greement.
 
 > The purpose of a CLA is to ensure that the guardian of a project's outputs has the necessary ownership or grants of rights over all contributions to allow them to distribute under the chosen licence.
 > [Wikipedia](http://en.wikipedia.org/wiki/Contributor_License_Agreement)
 
 You can read and sign our full Contributor License Agreement [here](http://clabot.confluent.io/cla).
+
+## Reporting Bugs and Issues
+
+Report bugs and issues by creating a new GitHub issue. Prior to creating an issue, please search
+through existing issues so that you are not creating duplicate ones. If a pull request exists that
+corresponds to the issue, mention this pull request on the GitHub issue.
+
+## Guidelines for Contributing Code, Examples, Documentation
+
+Code changes are submitted via a pull request (PR). When submitting a PR use the following
+guidelines:
+
+* Follow the style guide below
+* Add/update documentation appropriately for the change you are making. For more information, see the docs readme.
+* Non-trivial changes should include unit tests covering the new functionality and potentially integration tests.
+* Bug fixes should include unit tests and/or integration tests proving the issue is fixed.
+* Try to keep pull requests short and submit separate ones for unrelated features.
+* Keep formatting changes in separate commits to make code reviews easier and distinguish them from actual code changes.
+
```

Comment:
Should we add a guideline on targeting the earliest branch possible here or in the GitHub flow?

Reply:
nvm its already there below